zalando.yml

```diff
@@ -15,13 +15,13 @@ rules:
 
   # MUST always return JSON objects as top-level data structures [110]
   # => https://opensource.zalando.com/restful-api-guidelines/#110
-
   must-always-return-json-objects-as-top-level-data-structures:
     message: 'Top-level data structure must be an object'
     description: MUST always return JSON objects as top-level data structures [110]
     documentationUrl: https://opensource.zalando.com/restful-api-guidelines/#110
     severity: error
-    given: $.paths.*.*[responses,requestBody]..content..schema
+    # Only target JSON content types using a more specific JSONPath
+    given: $.paths.*.*[responses,requestBody]..content['application/json'].schema
```

Comment:
To conform to https://opensource.zalando.com/restful-api-guidelines/#114 we need to have a more percise selection of the content type.

I think as a short solution, we could use a JSONPath selection which tests for the absence of `application/(<optional-stuff)json`, and if not something like `application/json` or `application/x.zalando.cart+json;version=2` we don't apply this rule at all.

What do you think?

Reply:
You could inspier you with https://github.com/baloise-incubator/spectral-ruleset/blob/main/zalando.yml#L391

Reply:
Thanks for this hint. I added a small adaption. Is this fine like that?